command/applications/application_save.go

```diff
@@ -138,6 +139,56 @@ func (c *ApplicationSaveCommand) validApp(app map[string]interface{}) bool {
 	return providers != nil && name != "" && email != ""
 }
 
+// queryTask queries the task for the given ref and returns whether it was
```

Comment:
Given the name of this function & this comment I would recommend moving the sleep/retry logic elsewhere. This gives you a clean separation between a success call & a blocking wait until success/fail.

Reply:
SG.